connectors/sources/google_drive.py

```diff
@@ -195,22 +207,249 @@ async def _execute_api_call(self, resource, method, call_api_func, kwargs):
             raise
 
 
+class GoogleDriveClient(GoogleAPIClient):
+    """A google drive client to handle api calls made to Google Drive API."""
+
+    def __init__(self, json_credentials):
+        """Initialize the GoogleApiClient superclass.
+
+        Args:
+            json_credentials (dict): Service account credentials json.
+        """
+        super().__init__(
+            json_credentials=json_credentials,
+            api_name="drive",
+            api_version="v3",
+            scopes=["https://www.googleapis.com/auth/drive.readonly"],
+        )
+
+    async def ping(self):
+        return await self.api_call(resource="about", method="get", fields="kind")
+
+    async def get_drives(self):
+        """Fetch all shared drive (id, name) from Google Drive
+
+        Yields:
+            dict: Shared drive metadata.
+        """
+
+        async for drive in self.api_call_paged(
+            resource="drives",
+            method="list",
+            fields="nextPageToken,drives(id,name)",
+            pageSize=100,
+        ):
+            yield drive
+
+    async def retrieve_all_drives(self):
```

Comment:
Just curious: Why are some method names starting with `get` and others with `retrieve`?

Reply:
I might have been inspired by the naming conventions in the Workplace Search 😄 I agree that it needs revisiting